### PR TITLE
Update to AWS SDK for JavaScript (v3)

### PR DESCRIPTION
**Pull request by lyricnz**
Very very draft. https://github.com/Innablr/revolver/issues/17

### CODE REVIEW

**Comment by lyricnz** [comment]
Worth reviewing now @abukharov @alutman-innablr .  Only EC2 is properly converted (with the additional arg to the AWS client constructor).  Please let me know how to improve the code, and if I should proceed with the other clients (and/or make a type-specific factory).

Like
```
async function getEC2Client(remoteRole: string, region: string): Promise<EC2> {
  const creds = await assume.connectTo(remoteRole, region);
  const config = getAwsConfig(creds, region);
  return new EC2(remoteRole, region);
}
```

Or pass in an `acountConfig` and have the function get `.assumeRoleArn` and `.region` from it...?

File: drivers/ec2.ts [hunk 1]
```diff
@@ -201,10 +188,8 @@ class Ec2Driver extends DriverInterface {
     const inoperableStates = ['terminated', 'shutting-down'];
     logger.debug('EC2 module collecting account: %j', this.accountConfig.name);
 
-    const creds = await assume.connectTo(this.accountConfig.assumeRoleArn);
-
-    const ec2 = new EC2({ credentials: creds, region: this.accountConfig.region });
-    const autoscaling = new AutoScaling({ credentials: creds, region: this.accountConfig.region });
+    const ec2 = await getAwsClient(EC2, this.accountConfig.assumeRoleArn, this.accountConfig.region);
```
Comment (alutman-innablr) [inline drivers/ec2.ts]: Might as well replace with `getAwsClientForAccount` for both.

File: lib/awsConfig.ts [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+  const config = getAwsConfig(region, creds);
+  return new ctor(config);
+}
+
+// Helper function to get a specific AWS client for the given account
+async function getAwsClientForAccount<T>(ctor: Ctor<T>, accountConfig: any): Promise<T> {
+  return getAwsClient(ctor, accountConfig.assumeRoleArn, accountConfig.region);
+}
+
+export { getAwsConfig, getAwsClient, getAwsClientForAccount };
```
Comment (alutman-innablr) [inline lib/awsConfig.ts:49]: `getAwsClient` is only used in one spot in ec2 that could instead use `getAwsClientForAccount`. Might as well remove the export of `getAwsClient`.

**Review by alutman-innablr — CHANGES_REQUESTED** [review]
Minor note on a `awsConfig` function but otherwise looks good.

Note that this is still using the v2 "style" (e.g. `ec2.startInstances`) just with newer libraries. If/when we implement mocking we might to use v3 style  (e.g `ec2.send(new StartInstancesCommand(...)`). [Full details](https://docs.aws.amazon.com/AWSJavaScriptSDK/v3/latest/client/ec2/command/StartInstancesCommand/)

**Review by alutman-innablr — APPROVED** [review]

**Comment by lyricnz** [comment]
> Minor note on a `awsConfig` function but otherwise looks good.
> 
> Note that this is still using the v2 "style" (e.g. `ec2.startInstances`) just with newer libraries. If/when we implement mocking we might to use v3 style (e.g `ec2.send(new StartInstancesCommand(...)`). [Full details](https://docs.aws.amazon.com/AWSJavaScriptSDK/v3/latest/client/ec2/command/StartInstancesCommand/)

The new style feels awkward to me

**Pull request merged**